src/VecSim/algorithms/hnsw/hnsw_wrapper.cpp

```diff
@@ -130,11 +132,11 @@ VecSimQueryResult_List HNSWIndex::rangeQuery(const void *queryBlob, float radius
     void *timeoutCtx = nullptr;
     this->last_mode = RANGE_QUERY;
 
-    float normalized_blob[this->dim]; // This will be use only if metric == VecSimMetric_Cosine
-    if (this->metric == VecSimMetric_Cosine) {
-        // TODO: need more generic when other types will be supported.
-        memcpy(normalized_blob, queryBlob, this->dim * sizeof(float));
-        float_vector_normalize(normalized_blob, this->dim);
+    bool is_cosine = this->metric == VecSimMetric_Cosine;
+    float normalized_blob[this->dim *
```

Comment:
I don't understand the purpose of the code. Avoid stack-allocated array with multiplication?

Reply:
yes, the normalized blob size will be 0 if not cosine